### PR TITLE
Enable Churrito and Donut in genesis for e2e tests' local geth nodes.

### PR DESCRIPTION
**Pull request by oneeman**
### Description

With this PR, the custom genesis used by celotool for its local geth nodes will support the `churritoBlock` and `donutBlock` fields, which celo-blockchain uses to configure the activation blocks for these forks.  These are set to 0 for the e2e tests, so that they run with Churrito and Donut enabled.

The PR also adds support for the `CHURRITO_BLOCK` and `DONUT_BLOCK` environment variables to control the hard forks' activation in the context of the configuration `.env` files we use, but does not set these environment variables in any of the configuration files.

### Tested

- Confirmed that the e2e tests now have Churrito and Donut enabled at block 0.
- Checked that if I add values for `CHURRITO_BLOCK` and `DONUT_BLOCK` in the `.env` file (temporarily, without committing), and then use `celotool deploy initial testnet` then the resulting testnet's genesis.json file includes the corresponding `churritoBlock` and `donutBlock` values, whereas if I don't specify them in `.env` it doesn't have them.

### CODE REVIEW

File: packages/celotool/src/lib/genesis_constants.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ export const TEMPLATE = {
     constantinopleBlock: 0,
     petersburgBlock: 0,
     istanbulBlock: 0,
+    churritoBlock: 0,
```
Comment (mcortesi) [inline packages/celotool/src/lib/genesis_constants.ts]: this is used for every genesis block we generate. So for example whenever someone launches a new testnet.

For that i thin it should be configurable for such cases. Imagine you want to launch a blockchain where donut happens on block 1000.

Comment (oneeman) [reply packages/celotool/src/lib/genesis_constants.ts]: Good point.  I believe that should be easy enough to do.  I will add a commit.

Comment (oneeman) [reply packages/celotool/src/lib/genesis_constants.ts]: Updated the PR, so it no longer modifies TEMPLATE.

**Comment by oneeman** [comment]
@nategraf @mcortesi This PR is ready for review, please have a look.  I've updated the PR description to reflect the current content.  Here are a couple of things I didn't do and which I'd appreciate your thoughts on:

1. What should we do for `geth start`?  I'm not too familiar with its use-cases.  A couple of options are adding `churrito-block` and `donut-block` flags to it and having the default be no activation.  But that is putting additional burden on the user to remember to enable them in the future.  Alternatively, we could make the default 0, but I think then we'd need a separate flag (e.g. `no-churrito` to disable).  Also, if the default were 0, that would not work well for long-lived networks like mainnet, alfajores or baklava (though I'm not sure if `geth start` is really used with them).

2. Should we set `CHURRITO_BLOCK` and `DONUT_BLOCK` in any of the `.env` files?  I'm not too clear on what each of the env files is for.  Of course, for mainnet, alfajores, and baklava we don't need to, because their activation blocks will be set in celo-blockchain.  But I don't know about the various other ones we have.

**Review by nategraf — APPROVED** [review]
Looks good to me!

On the topic of adding these to the `.env` files, it seems like a good idea to add it to `.env`, but not `.env.mainnet` or the respective files for Alfajores and Baklava. We don't really use those files anymore given that we are not redeploying those networks. If we do have a time where we need to redeploy either Alfajores or Baklava we my decide to update these then.

**Comment by oneeman** [comment]
> Looks good to me!
> 
> On the topic of adding these to the `.env` files, it seems like a good idea to add it to `.env`, but not `.env.mainnet` or the respective files for Alfajores and Baklava. We don't really use those files anymore given that we are not redeploying those networks. If we do have a time where we need to redeploy either Alfajores or Baklava we my decide to update these then.

Sounds good!  Since this has been merged, I created a new PR with the `.env` change and adding support to celotool's `geth start` command using `coerce` as you suggested.  It's #6574.

**Pull request merged**